common/cah-model.h

```diff
@@ -0,0 +1,65 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <chrono>
+#include <atomic>
+
+#include <librealsense2/rs.hpp>
+
+namespace rs2
+{
+    class ux_window;
+    class viewer_model;
+    class device_model;
+
+    // Helper class for setting , starting and checking a timeout.
+    // Inner units are [ms] , API units are seconds.
+    class timeout
+    {
+    public:
+        // Start timer from current time
+        // Can be called with std::chrono::hours/minutes/seconds/milliseconds/microseconds/nanoseconds type
+        void start(std::chrono::seconds timeout)
+        {
+            using namespace std::chrono;
+            start_time_ms = duration_cast<milliseconds>(high_resolution_clock::now().time_since_epoch()).count();
+            timeout_delay_ms = duration_cast<milliseconds>(timeout).count();
+        }
+
+        // Check if timeout expired
+        bool has_expired()
+        {
+            using namespace std::chrono;
+            auto curr_time = duration_cast<milliseconds>(high_resolution_clock::now().time_since_epoch()).count();
+            return curr_time - start_time_ms >= timeout_delay_ms;
+        }
+
+        timeout() : timeout_delay_ms(), start_time_ms() {};
```

Comment:
If they're undefined, so is the behavior before start()... can we make this more reliable??

Reply:
I also prefer the ctor to be first in the class decl, you like it in the end, or just by accident?

Reply:
> If they're undefined, so is the behavior before start()... can we make this more reliable??

You are right, it started as a different type and changed.
Will set to 0.

Reply:
> I also prefer the ctor to be first in the class decl, you like it in the end, or just by accident?

Accepted, no special thought on that, Could be mode readable at first, will be removed.